src/pages/ServiceGraph/ServiceGraphPage.tsx

```diff
@@ -36,15 +36,19 @@ const cytscapeGraphStyle = style({
 
 export default class ServiceGraphPage extends React.PureComponent<ServiceGraphPageProps> {
   private pollTimeoutRef?: number;
+  private mounted: boolean;
   constructor(props: ServiceGraphPageProps) {
     super(props);
+    this.mounted = false;
   }
 
   componentDidMount() {
+    this.mounted = true;
     this.scheduleNextPollingInterval(0);
   }
 
   componentWillUnmount() {
+    this.mounted = false;
     this.removePollingIntervalTimer();
```

Comment:
> componentWillUnmount() is invoked immediately before a component is unmounted and destroyed. Perform any necessary cleanup in this method, such as **invalidating timers**, canceling network requests, or cleaning up any subscriptions that were created in componentDidMount().
>
> You should not call setState() in componentWillUnmount() because the component will never be re-rendered. Once a component instance is unmounted, it will never be mounted again.

Reply:
I see the problem in removePollingInterval.
I'm using a promise there, the problem happens when the timer fired, but we are "waiting" on the promise.
The right thing to do is to cancel that promise...
Check it out here: https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html

Reply:
I read that page before opening the PR and I though this case it wasn't falling in the antipattern case, because we are not calling `setState`.
Anyway. I've updated the PR.

Reply:
> using isMounted() is a code smell because the only reason you would check is because you think you might be holding a reference after the component has unmounted.

Reply:
touché